cmake/match.py

```diff
@@ -87,6 +88,12 @@ def main():
             )
         )
 
+    # Every fail has a single corresponding match line but if there are multiple
+    # devices being tested there will be multiple input lines with the same failure
+    # message. To avoid matching mismatch, remove lines that differ only by device ID.
+    input_lines = [re.sub(r'ID[0-9]ID', 'X', line) for line in input_lines]
```

Comment:
we use this for more than just test match files. Maybe you can remove the duplicate lines earlier, like in run_cts.py?

Reply:
That makes sense. Done.